dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -518,11 +581,21 @@ void InterpreterSelectQuery::executeImpl(Pipeline & pipeline, const BlockInputSt
 
     AnalysisResult expressions;
 
+    if (storage && context.hasUserProperty(storage->getDatabaseName(), storage->getTableName(), "filter"))
+    {
+        ExpressionActionsPtr actions;
+        generateFilterActions(actions, storage, context, required_columns);
```

Comment:
Don't we calc it again in `analyzeExpressions` function? Can't we use it from this call?

Reply:
I got rid of the second invocation inside `analyzeExpressions()`